api/src/modules/materials/material.entity.ts

```diff
@@ -112,25 +112,28 @@ export class Material extends TimestampedBaseEntity {
   @JoinColumn({ name: 'layerId' })
   layer: Layer;
 
-  /**
-   * @debt: This columns should not be nullable but is necessary for the time being
-   * to load material data since we have no layer data yet
-   */
   @ApiProperty()
   @Column()
   layerId!: string;
 
+  @Column({ type: 'text', nullable: true })
+  datasetId: string;
+
+  @OneToOne(() => H3Data, (h3grid: H3Data) => h3grid.id, {
+    nullable: true,
+    eager: true,
+  })
+  @JoinColumn()
+  producer: H3Data;
+  @Column({ nullable: true })
+  producerId: string;
+
   @OneToOne(() => H3Data, (h3grid: H3Data) => h3grid.id, {
     nullable: true,
     eager: true,
   })
-  @JoinColumn({ name: 'h3GridId' })
-  h3Grid: H3Data;
-  /**
-   * @note: Although there should not be any material with no h3GridId, this needs to be nullable
-   * since materials needs to exist before the h3 script takes care of adding these IDs to
-   * each material
-   */
+  @JoinColumn()
+  harvest: H3Data;
   @Column({ nullable: true })
-  h3GridId!: string;
+  harvestId: string;
```

Comment:
meta question: why "harvest"? what if the material is a mineral or something else?

Reply:
meta answer: naming does not reference the 'kind' of the material, but the type of the dataset, which we use then to determine which kind of calculus we need to do:
-Material Map -> producer
-Risk-Map -> harvest

Also, it is required by data to link existing materials to magically imported h3 without the API being involved